packages/next/src/lib/recursive-delete.ts

```diff
@@ -90,33 +90,16 @@ export async function recursiveDeleteSyncWithAsyncRetries(
   await Promise.all(
     result.map(async (part: Dirent) => {
       const absolutePath = join(dir, part.name)
-
-      // readdir does not follow symbolic links
-      // if part is a symbolic link, follow it using stat
-      let isDirectory = part.isDirectory()
-      const isSymlink = part.isSymbolicLink()
-
-      if (isSymlink) {
-        const linkPath = fs.readlinkSync(absolutePath)
-
-        try {
-          const stats = fs.statSync(
-            isAbsolute(linkPath)
-              ? linkPath
-              : join(dirname(absolutePath), linkPath)
-          )
-          isDirectory = stats.isDirectory()
-        } catch {}
-      }
-
       const pp = join(previousPath, part.name)
       const isNotExcluded = !exclude || !exclude.test(pp)
 
       if (isNotExcluded) {
-        if (!isSymlink && isDirectory) {
+        // Note: readdir does not follow symbolic links, that's what we want
```

Comment:
we want that because we never write links into .next, or because the intention is to delete the link not the destination? can you clarify?

Reply:
The intention is to delete the link and not the destination. See https://github.com/vercel/next.js/pull/82191

Most `rm -rf` type algorithms don't follow symlinks for this reason.